### PR TITLE
Make it compatible with Ruby's BasicObject

### LINKED ISSUE #20: Runtime error with latest Hanami
When installing this gem alongside the latest Hanami version, after creating a `.json.jbuilder` file, this pops up when reloading the browser:

```
Boot Error

Something went wrong while loading /home/x/hanami/bookshelf/config.ru

NameError: undefined local variable or method `binding' for #<Hanami::View::Rendering::Scope:0x00558771791078>

/home/x/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/hanami-view-0.8.0/lib/hanami/view/rendering/layout_scope.rb:270:in `method_missing'
/home/x/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/hanami-view-0.8.0/lib/hanami/view/rendering/scope.rb:72:in `method_missing'
```

### PR DESCRIPTION
**Pull request by jodosha**
Hi, I'm the author of Hanami, the web framework for Ruby.

Our gem `hanami-view` uses `tilt` to render templates. During the rendering process, the view passes to the `Tilt::Template` a scope which inherits from Ruby's `BasicObject`.

This decision was taken to avoid conflicts with methods defined by Ruby's `Object`, which had the result of shadowing methods in the scope. This is one example of the clash: https://github.com/hanami/view/issues/28

---

There are two devs who reported a bug both in [your repo](https://github.com/anthonator/tilt-jbuilder/issues/20) and in [ours](https://github.com/hanami/view/issues/124) as well, because our gems aren't compatible.

This PR aims to fix these problems, by making `tilt-jbuilder` compatible with `BasicObject`, and by consequence with Hanami.

Please consider to merge this, I believe that `tilt-jbuilder` can benefit from this, regardless the Hanami compatibility. Again, see https://github.com/hanami/view/issues/28 for a concrete example.

Thanks in advance for your time.

---

Fix: #20

### CODE REVIEW

File: lib/tilt/jbuilder.rb [hunk 1]
```diff
@@ -103,9 +103,10 @@ def evaluate(scope, locals, &block)
     private
     def set_locals(locals, scope, context)
       view_path = options[:view_path]
-      scope.send(:instance_variable_set, '@_jbuilder_view_path', view_path)
-      scope.send(:instance_variable_set, '@_jbuilder_locals', locals)
-      scope.send(:instance_variable_set, '@_tilt_data', data)
+      scope.instance_exec(binding) {
+        @_jbuilder_view_path = view_path
+        @_jbuilder_locals = locals
+      }
```
Comment (anthonator) [inline lib/tilt/jbuilder.rb:109]: What's the purpose of this change? Is this for `BasicObject` compatibility or an unrelated changed?

**Comment by anthonator** [comment]
@jodosha thanks for submitting this! Do you know if the other Tilt templates use `BasicObject` in this manner as well?

File: lib/tilt/jbuilder.rb (hunk 1 above)
Comment (jodosha) [reply lib/tilt/jbuilder.rb:109]: It's related because `BasicObject` doesn't respond to `#send`.

Comment (jodosha) [reply lib/tilt/jbuilder.rb:109]: Semi-related: I removed the line that was setting `data` because it wasn't clear to me where `data` was previously instantiated, and because the tests were still passing without it.

**Comment by jodosha** [comment]
@anthonator My pleasure to make gems to work together. 😉 

I don't know if other Tilt templates use `BasicObject`, but we use it in Hanami since three years now with ERb/HAML/Slim and it works just great.

The suggestion to use `BasicObject` came from one of the Tilt maintainers: https://github.com/hanami/view/issues/28#issuecomment-54128647

File: lib/tilt/jbuilder.rb (hunk 1 above)
Comment (anthonator) [reply lib/tilt/jbuilder.rb:109]: Unless you feel like you have a good reason not to would you mind adding `data` back in? Removing it seems unrelated to this PR and I'd rather not risk introducing a bug.

**Comment by anthonator** [comment]
Other than an issue I brought up above I think this looks good.

Would you mind adding some information to the README on how to use `tilt-jbuilder` with Hanami?

File: lib/tilt/jbuilder.rb (hunk 1 above)
Comment (jodosha) [reply lib/tilt/jbuilder.rb:109]: @anthonator I see your concern and totally agree with it.

But if add that `data` back as third line, it raises a `NoMethodError`. Which leads us to the initial issue: I don't understand where is defined. 😉

**Comment by jodosha** [comment]
@anthonator sure thing. 😄
